modules/core/src/v2/coins/sol.ts

```diff
@@ -27,10 +29,7 @@ export interface TransactionFee {
 export type SolTransactionExplanation = TransactionExplanation;
 
 export interface ExplainTransactionOptions {
-  txHex?: string;
-  halfSigned?: {
-    txHex: string;
-  };
```

Comment:
seems like we'd want this? see that this is included for some points, and not then not included for others. where it is, have the following code under `explainTransaction`

```
      const txHex = params.txHex || (params.halfSigned && params.halfSigned.txHex);
      if (!txHex || !params.feeInfo) {
        throw new Error('missing explain tx parameters');
      }
```

Reply:
@0xabrarhussain imo i don't think this is necessary. i'm not sure why we need to have clients differentiate between an unsigned/halfsigned tx when calling this as we won't do anything differently.

some examples where we don't differentiate:
```
// XLM
interface ExplainTransactionOptions {
    txBase64: string;
}

// EOS
interface ExplainTransactionOptions {
    transaction: {
        packed_trx: string;
    };
    headers: EosTransactionHeaders;
}
```

Reply:
looking at it, I think it's coin specific. I see that some coins have a `txhex` in the half-signed object while others don't

Reply:
@alebusse good to know, but from the client's perspective, it should matter if the call `explainTransaction` using an unsigned, half signed, or fully signed hex.

looking at the param, the client already knows if the transaction is half-signed or not. otherwise, they wouldn't know whether to put their tx string in `halfSigned` or not.

e.g.
```
// we expect the two to be equivalent
const halfSigned = sol.explainTransaction({ halfSigned: { txBase64: halfSignedBase64 }});
const alsoHalfSigned = sol.explainTransaction({ txBase64: halfSignedBase64 });
```

Reply:
If tx initially is single sig. Why we should have halfsigned tx? For this will be only signed or unsigned for this initial case. We should discard halfsigned tx for Solana.
OVC will be used for initial case that we are supporting only single sig?
This discussion should be taken when we encourage the TSS approach and Cold Wallet Integration.

Reply:
agreed - we can add it if/when we have a need to differentiate between a half-signed tx or not.